test/e2e/e2e_general_cli_test.go

```diff
@@ -83,12 +83,13 @@ func testGeneralCliStuff(t *testing.T, terraformOptions *terraform.Options, keyP
 	assert.NotEqual(t, string(output), "UnknownVersion", "Zarf version should not be the default value")
 
 	// Test for expected failure when given a bad component input
-	output, err = ssh.CheckSshCommandE(t, publicHost, fmt.Sprintf("cd /home/%s/build && ./zarf init --components management,foo,logging", username))
+	output, err = ssh.CheckSshCommandE(t, publicHost, fmt.Sprintf("cd /home/%s/build && ./zarf init --host 127.0.0.1 --components management,foo,logging", username))
 	require.Error(t, err, output)
 
 	// Test for expected failure when given invalid hostnames
-	output, err = ssh.CheckSshCommandE(t, publicHost, fmt.Sprintf("cd /home/%s/build && ./zarf init --host bad!hostname", username))
-	require.Error(t, err, output)
+	// output, err = ssh.CheckSshCommandE(t, publicHost, fmt.Sprintf("cd /home/%s/build && ./zarf init --host bad!hostname", username))
```

Comment:
This PR recursively asks the user for the hostname if an invalid option is provided which would hang this test. Probably should have removed instead of just commenting it out but I wanted to see if anyone had opinions/thoughts on how we could still test this.

Reply:
I think it's better to prompt again rather than just abort if it's an active user prompt, but if they pass `--host` with an invalid entry it should abort.  For now if it's easier we can just maintain the existing abort behavior.

Reply:
I just ran across this. Running `./zarf init --confirm --components management,gitops-service --host localhost` prompts me for a different host rather than throwing a fatal

Reply:
`./zarf init --confirm --components management,gitops-service --host 127.0.0.1` also get stuck waiting for a user prompt.

(I'm fairly sure you guys already know about this and are already working on it, just making sure I'm documenting the things I do as I test)

Reply:
For the second comment, I added a `--generate` flag so we wouldn't get stuck waiting for a user prompt. 

I haven't addressed the first comment though, Jeff messaged me saying he was going to rewrite the cert/hostname logic so I was going to wait for him to push thinking he was going to take care of it.

Reply:
Got it, so we are now expecting either `--generate` or `--server-crt` and `--server-key`? That works for now. Later we should add a check that fails if `--confirm` is set but `--generate` or `--server-crt --server-key` isn't.

Reply:
Should we name it `--generate-ca` or something? `--generate` doesn't seem quite descriptive enough (generate what?)

Reply:
hey sorry I'm about done rewriting all this, messaged @YrrepNoj should have put it in here.

Reply:
I actually threw away the generate flag as it's duplicate of providing the cert paths

Reply:
sorry @RothAndrew

Reply:
Not a problem. I'll go through and undo adding the `--generate` flag to the docs and stuff, should only take a couple of minutes

Reply:
these issues are addressed in the last commit